src/server.c

```diff
@@ -17,1657 +17,609 @@
-        // We want to set the user name only if it is not already set and the selected sasl mechanism is EXTERNAL
-        if (mech && strcmp(mech, MECH_EXTERNAL) == 0) {
-            const char *user_id = transport_get_user(conn, tport);
-            if (user_id)
-                conn->user_id = user_id;
-        }
-    }
-}
-
-
-qd_error_t qd_entity_refresh_sslProfile(qd_entity_t* entity, void *impl)
-{
+qd_error_t qd_entity_refresh_sslProfile(qd_entity_t* entity, void* impl) {
     return QD_ERROR_NONE;
 }
 
-qd_error_t qd_entity_refresh_authServicePlugin(qd_entity_t* entity, void *impl)
-{
+qd_error_t qd_entity_refresh_authServicePlugin(qd_entity_t* entity, void* impl) {
     return QD_ERROR_NONE;
 }
 
+qd_server_t* qd_server(qd_dispatch_t* qd, int thread_count, const char* container_name, const char* sasl_config_path,
+                       const char* sasl_config_name) {
+    // Initialize const members. Zero initialize all others.
 
-static qd_error_t listener_setup_ssl(qd_connection_t *ctx, const qd_server_config_t *config, pn_transport_t *tport)
-{
-    pn_ssl_domain_t *domain = pn_ssl_domain(PN_SSL_MODE_SERVER);
-    if (!domain) return qd_error(QD_ERROR_RUNTIME, "No SSL support");
+    qd_server_t tmp = {.thread_count = thread_count};
+    // XXX This type contructor is different from what I see elsewhere.
+    qd_server_t* server = NEW(qd_server_t);
 
-    // setup my identifying cert:
-    if (pn_ssl_domain_set_credentials(domain,
-                                      config->ssl_certificate_file,
-                                      config->ssl_private_key_file,
-                                      config->ssl_password)) {
-        pn_ssl_domain_free(domain);
-        return qd_error(QD_ERROR_RUNTIME, "Cannot set SSL credentials");
+    if (!server) {
+        return NULL;
     }
 
-    // for peer authentication:
-    if (config->ssl_trusted_certificate_db) {
-        if (pn_ssl_domain_set_trusted_ca_db(domain, config->ssl_trusted_certificate_db)) {
-            pn_ssl_domain_free(domain);
-            return qd_error(QD_ERROR_RUNTIME, "Cannot set trusted SSL CA" );
-        }
-    }
+    memcpy(server, &tmp, sizeof(tmp));
 
-    if (config->ssl_ciphers) {
-        if (pn_ssl_domain_set_ciphers(domain, config->ssl_ciphers)) {
-            pn_ssl_domain_free(domain);
-            return qd_error(QD_ERROR_RUNTIME, "Cannot set ciphers. The ciphers string might be invalid. Use openssl ciphers -v <ciphers> to validate");
-        }
-    }
+    server->qd                   = qd;
+    server->log_source           = qd_log_source("SERVER");
+    server->protocol_log_source  = qd_log_source("PROTOCOL");
+    server->container_name       = container_name;
+    server->sasl_config_path     = sasl_config_path;
+    server->sasl_config_name     = sasl_config_name;
+    server->proactor             = pn_proactor();
+    server->container            = NULL;
+    server->start_context        = 0;  // XXX Should be NULL?
```

Comment:
Probably, since this is typed as a void pointer.

Reply:
In C++, `clang-tidy` pushes towards using the (cpp only) nullptr. Not sure what cppcheck does. I guess that C linters, if any was used, would, by default, push for using NULL.